fundingmanager.go

```diff
@@ -119,7 +119,7 @@ type fundingManager struct {
 	// channelReservations is a map which houses the state of all pending
 	// funding workflows.
 	resMtx             sync.RWMutex
-	activeReservations map[int32]pendingChannels
+	activeReservations map[string]pendingChannels
```

Comment:
Rather than using a `string` here (which requires a series of heap allocations in order to copy it over or mutate) we can instead use a fixed size `[33]byte` for tracking the reservations of a particular peer. 

We might also want to add a new type to make the definition a bit more readable: 
```go
type reservationClient [33]byte
```

Feel free to pick a better name for the type itself.